taichi/runtime/llvm/runtime.cpp

```diff
@@ -1741,6 +1744,40 @@ i32 wasm_materialize(Context *context) {
   return (i32)(size_t)context->runtime->roots[0];
 }
 
+void wasm_set_print_buffer(Context *context, Ptr buffer) {
+  context->runtime->wasm_print_buffer = buffer;
+}
+
+void wasm_print_i32(Context *context, i32 value) {
+  Ptr buffer = context->runtime->wasm_print_buffer;
+  if (buffer == nullptr)
+    return;
+  i32 total_cnt = ((i32 *)buffer)[0]++;
+  i32 print_pos = total_cnt % (1024 * 1024);
+  ((i32 *)buffer)[print_pos * 2 + 1] = 0;  // 0 for i32
+  ((i32 *)buffer)[print_pos * 2 + 2] = value;
+}
+
+void wasm_print_f32(Context *context, f32 value) {
+  Ptr buffer = context->runtime->wasm_print_buffer;
+  if (buffer == nullptr)
+    return;
+  i32 total_cnt = ((i32 *)buffer)[0]++;
+  i32 print_pos = total_cnt % (1024 * 1024);
```

Comment:
nit: name this to `kWasmPrintBufferSize`

Reply:
https://github.com/taichi-dev/taichi/pull/2910/commits/147a2505cb388108b2c99833b12d95f04372b507